artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -4218,7 +4219,7 @@ public QueueConfiguration getQueueConfiguration() {
          .setAutoCreated(autoCreated);
    }
 
-   protected static class ConsumerHolder<T extends Consumer> implements PriorityAware {
+   protected class ConsumerHolder<T extends Consumer> implements PriorityAware {
```

Comment:
Rather than make it non-static and add a new reference to the parent queue object, it seems like all the Consumer impls passsed to addConsumer already have getQueue(). Could that be promoted to the interface and used instead to get the name for the toString?